theano/tensor/nnet/abstract_conv.py

```diff
@@ -357,13 +374,185 @@ def conv2d_grad_wrt_weights(input,
         version until it is released.
 
     """
+
+    input = as_tensor_variable(input)
+    output_grad = as_tensor_variable(output_grad)
+
+    # checking the type of filter_shape
+    for dim in [0, 1]:
+        assert isinstance(filter_shape[dim], (theano.tensor.TensorConstant,
+                                              int, type(None)))
+    for dim in [2, 3]:
+        assert isinstance(filter_shape[dim], (theano.tensor.TensorVariable,
+                                              theano.tensor.TensorConstant,
+                                              int))
+
+    # setting the last two dimensions of filter_shape to None, if
+    # the type of these dimensions is TensorVariable.
+    numerical_filter_shape = filter_shape
+    for dim in [2, 3]:
+        if isinstance(filter_shape[dim], theano.tensor.TensorVariable):
+            numerical_filter_shape[dim] = None
+
     gradWeight_op = AbstractConv2d_gradWeights(imshp=input_shape,
-                                               kshp=filter_shape,
+                                               kshp=numerical_filter_shape,
                                                border_mode=border_mode,
                                                subsample=subsample,
                                                filter_flip=filter_flip)
 
-    return gradWeight_op(input, output_grad, filter_shape)
+    return gradWeight_op(input, output_grad, filter_shape[:-2])
+
+
+def bilinear_kernel_2D(ratio, normalize=True):
+    """ This function gets a ratio
+    and returns the normalized 2D filters that can
+    be applied to any image to upsample it by that ratio
+    using bilinear interpolation.
+
+    :type ratio: python integer or symbolic theano
+                 variable (Constant or Scalar)
+    : param ratio: the ratio by which an image will
+        be upsampled by the returned filters in the 2D space.
+
+    :type normalize: bool
+    : param normalize: indicates whether to normalize the
+    kernel or not. Default is True.
+
+    returns: the 2D kernels that can be applied to
+             any given image to upsample it by the indicated ratio
+             using bilinear interpolation.
+    """
+
+    hkern = bilinear_kernel_1D(ratio=ratio, normalize=normalize).dimshuffle('x', 0)
+    vkern = bilinear_kernel_1D(ratio=ratio, normalize=normalize).dimshuffle(0, 'x')
+    kern = hkern * vkern
+    return kern
+
+
+def bilinear_kernel_1D(ratio, normalize=True):
+    """ This function gets a ratio
```

Comment:
This docstring has a weird formatting. 
Also, according to [PEP257](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings), the multi-line docstring should consist of a (**i.e., one**) summary line, followed by a blank line, followed by a more elaborate description.